schema/cached.go

```diff
@@ -0,0 +1,69 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package schema
+
+import (
+	"context"
+	"time"
+
+	"github.com/conduitio/conduit-commons/schema"
+	"github.com/conduitio/conduit-processor-sdk/pprocutils"
+	"github.com/twmb/go-cache/cache"
+)
+
+func newCachedSchemaService(service pprocutils.SchemaService) *cachedSchemaService {
```

Comment:
@lovromazgon right. I should have explained it better. Initially, when I reviewed it i thought that if this only used by the test, then it should be moved to the test file (i.e.: an utility to use there), but if it stayed in `cached.go` made me think this was going to be used elsewhere (hence it should be exported).

Reply:
I wouldn't have a problem moving this to a test file, but that won't work 😅 So this is used to wrap the in-memory schema service and assigned to the global variable. _We know_ that the default schema service is used only in tests and that this global var is overwritten in actual processors, but Go doesn't know that. So we can't really have this code in a test, because then we can't use it to instantiate the global variable.

Or maybe I'm again misunderstanding the suggestion?

Reply:
@lovromazgon ah! I missed this https://github.com/ConduitIO/conduit-processor-sdk/blob/5f1ccda8b5fa36ea021dc1ccd0724d32975efe1b/schema/schema.go#L27